scipy/spatial/transform/_rigid_transform.py

```diff
@@ -1413,15 +1201,32 @@ def __getitem__(self, indexer):
         array([[0., 0., 0.],
                [2., 0., 0.]])
         """
-        if self._single:
+        if self.single:
             raise TypeError("Single transform is not subscriptable.")
 
-        # Convert memoryview to numpy array before indexing:
-        arr = np.asarray(self._matrix)
-        return self.__class__(arr[indexer], copy=False)
-
-    @cython.embedsignature(True)
-    def __setitem__(self, indexer, value):
+        is_array = isinstance(indexer, type(self._matrix))
+        # TODO: Getting xp on every call may be expensive. Check if we can make access
+        # to xp more efficient. Should we store a self._xp attribute? What about pickle?
+        xp = array_namespace(self._matrix)
+        # Masking is only specified in the Array API when the array is the sole index
+        # This special case handling is necessary to support boolean indexing and
+        # integer array indexing with take (see 
+        # https://github.com/data-apis/array-api/pull/900#issuecomment-2674432480)
+        # Ideally we would converge to [indexer, ...] indexing, but this is not
+        # supported for now.
+        if is_array and indexer.dtype == xp.bool:
+            return RigidTransform(self._matrix[indexer], normalize=False)
+        if is_array and (indexer.dtype == xp.int64 or indexer.dtype == xp.int32):
+            return RigidTransform(
+                xp.take(self._matrix, indexer, axis=0), normalize=False
+            )
```

Comment:
The spec says that we can only use `take` when `indexer.ndim == 1`. I guess we either use `xpx.atleast_nd`, or add a separate case for empty indexing.

nit:

```suggestion
        if is_array and indexer.dtype == xp.bool:
            return RigidTransform(self._matrix[indexer], normalize=False)
        if is_array and xp.isdtype(indexer.dtype, "integral"):
            return RigidTransform(
                xp.take(self._matrix, indexer, axis=0), normalize=False
            )
```

Reply:
Strange that this doesn't fail locally on 64bit then. Is there a simple way to run the 32bit tests locally?

Reply:
This is a case where NumPy implements a superset of the spec, but not quite enough for us here.

This would fail once we start testing with array-api-strict:

```python
In [1]: import array_api_strict as xp

In [2]: x = xp.arange(5)

In [3]: idx = xp.empty((), dtype=xp.int64)

In [4]: xp.take(x, idx)
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
Cell In[4], line 1
----> 1 xp.take(x, idx)

File ~/Library/Caches/rattler/cache/cached-envs-v0/ipython-eb5c9ad40873f671/lib/python3.13/site-packages/array_api_strict/_indexing_functions.py:25, in take(x, indices, axis)
     23     raise TypeError("Only integer dtypes are allowed in indexing")
     24 if indices.ndim != 1:
---> 25     raise ValueError("Only 1-dim indices array is supported")
     26 if x.device != indices.device:
     27     raise ValueError(f"Arrays from two different devices ({x.device} and {indices.device}) can not be combined.")

ValueError: Only 1-dim indices array is supported
```

Reply:
On my end, this works just fine:

```python
import array_api_strict as xp

x = xp.arange(5)

idx = xp.asarray([], dtype=xp.int64)
y = xp.take(x, idx)  # empty((0,), dtype=array_api_strict.int64)
```
which is exactly what we are using in the tests. How do the tests work for 32bit systems? Do they just set the default of non-explicit types to the 32bit version? If so, explicitly requesting an int64 indices array might be the issue if they cannot be used as index array on 32bit systems. This would also explain the attempted cast from i64 to i32, which then throws.

Reply:
ah sorry, I guess `xp.asarray([], dtype=xp.int64)` counts as 1-d

> How do the tests work for 32bit systems?

As far as I know, we don't do anything special. It is just about the default NumPy behaviour on such a system.

> explicitly requesting an i64 indices array might be the issue if they cannot be used as index array on 32bit systems. This would also explain the attempted cast from i64 to i32, which then throws.

Yes, that sounds right.

Reply:
Resolved by using i32 in tests